File: src/main/java/com/codeperfection/shipit/repository/ProductRepository.java
package com.codeperfection.shipit.repository;

import com.codeperfection.shipit.entity.Product;
import com.codeperfection.shipit.entity.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface ProductRepository extends JpaRepository<Product, UUID> {

    List<Product> findByUserAndIsActiveTrue(User user);

    Page<Product> findByUserAndIsActiveTrue(User user, Pageable pageable);

    Optional<Product> findByUuidAndUser(UUID uuid, User user);
}

File: src/main/kotlin/com/codeperfection/shipit/repository/ProductRepository.kt
package com.codeperfection.shipit.repository

import com.codeperfection.shipit.entity.Product
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface ProductRepository : JpaRepository<Product, UUID> {

    fun findByUserIdAndIsActiveTrue(userId: UUID, pageable: Pageable): Page<Product>

    fun findAllByUserIdAndIsActiveTrueAndCountInStockGreaterThan(userId: UUID, minCountInStock: Int): List<Product>

    fun findByIdAndUserIdAndIsActiveTrue(productId: UUID, userId: UUID): Product?
}
